pkg/storage/storage.go

```diff
@@ -325,15 +362,45 @@ func (s *storageModule) CreateFilesystem(name string, size uint64, poolType pkg.
 func (s *storageModule) ReleaseFilesystem(name string) error {
 	log.Info().Msgf("Deleting volume %v", name)
 
-	for idx := range s.volumes {
-		filesystems, err := s.volumes[idx].Volumes()
+	for _, pool := range s.pools {
+		if _, mounted := pool.Mounted(); !mounted {
+			continue
+		}
+
+		filesystems, err := pool.Volumes()
 		if err != nil {
 			return err
 		}
-		for jdx := range filesystems {
-			if filesystems[jdx].Name() == name {
-				log.Debug().Msgf("Removing filesystem %v in volume %v", filesystems[jdx].Name(), s.volumes[idx].Name())
-				return s.volumes[idx].RemoveVolume(filesystems[jdx].Name())
+		for _, fs := range filesystems {
+			if fs.Name() == name {
+				log.Debug().Msgf("Removing filesystem %v in volume %v", fs.Name(), pool.Name())
+				err = pool.RemoveVolume(fs.Name())
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		// After releasing a filesystem by name, there is a chance there is going to be
+		// no reserved space on a volume, in this case we want to shutdown the disk in order to save power
+		volumes, err := pool.Volumes()
+		if err != nil {
+			log.Error().Err(err).Msgf("Failed to retrieve subvolumes on pool %s", pool.Name())
+		}
+		log.Debug().Msgf("Pool %s is has: %d subvolumes", pool.Name(), len(volumes))
+
+		if len(volumes) == 0 {
+			err = pool.UnMount()
+			if err != nil {
+				log.Error().Err(err).Msgf("Failed to unmount volume %s", pool.Name())
+				return err
+			}
+			for _, device := range pool.Devices() {
```

Comment:
I think it's better if you introduce a `shutdownPool` method that takes a pool, and shutdown all pool devices. so you don't have this code everywhere.

Reply:
There is a Shutdown method on a pool now, this makes it easier